lms/djangoapps/course_blocks/transformers/library_content.py

```diff
@@ -0,0 +1,176 @@
+"""
+Content Library Transformer, used to filter course structure per user.
+"""
+import json
+from courseware.models import StudentModule
+from openedx.core.lib.block_cache.transformer import BlockStructureTransformer
+from xmodule.library_content_module import LibraryContentModule
+from xmodule.modulestore.django import modulestore
+from eventtracking import tracker
+
+
+class ContentLibraryTransformer(BlockStructureTransformer):
+    """
+    Content Library Transformer Class
+    """
+    VERSION = 1
+
+    @classmethod
+    def _get_selected_modules(cls, user, course_key, block_key):
+        """
+        Get list of selected modules in a library,
+        for user.
+
+        Arguments:
+            user (User)
+            course_key (CourseLocator)
+            block_key (BlockUsageLocator)
+
+        Returns:
+            list[modules]
+        """
+        return StudentModule.objects.filter(
+            student=user,
+            course_id=course_key,
+            module_state_key=block_key,
+            state__contains='"selected": [['
+        )
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        Computes any information for each XBlock that's necessary to execute
+        this transformer's transform method.
+
+        Arguments:
+            block_structure (BlockStructureCollectedData)
+        """
+        block_structure.request_xblock_fields('mode')
+        block_structure.request_xblock_fields('max_count')
+        store = modulestore()
+
+        # needed for analytics purposes
+        def summarize_block(usage_key):
+            """ Basic information about the given block """
+            orig_key, orig_version = store.get_block_original_usage(usage_key)
+            return {
+                "usage_key": unicode(usage_key),
+                "original_usage_key": unicode(orig_key) if orig_key else None,
+                "original_usage_version": unicode(orig_version) if orig_version else None,
+            }
+
+        # For each block check if block is library_content.
+        # If library_content add children array to content_library_children field
+        for block_key in block_structure.topological_traversal():
+            xblock = block_structure.get_xblock(block_key)
+            block_structure.set_transformer_block_data(block_key, cls, 'content_library_children', [])
```

Comment:
Is `content_library_children` needed?  Can you get that information from the `block_structure` instead?
Also, other transformers may possibly remove children because of other access issues so we should get the latest list from the `block_structure`.